hack/setup-test-environment.sh

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
```

Comment:
This seems to duplicate a lot of `setup-dev-environment.sh`. Maybe on the next iteration we can combine the two?

Reply:
Some of it is, there could be a `environment.sh` that has the shared bits in -- I didn't think it was worth it right now. This one uses a different Kustomize directory (development vs test) to install Chains in addition to Tekton and the image repository, doesn't build/push the images and waits on Chains in addition to Tekton & image repository to be available.